fuzz/fuzz_targets/fuzz_common.rs

```diff
@@ -277,20 +277,50 @@ pub fn compare_result(
 
 pub fn generate_random_string(max_length: usize) -> String {
     let mut rng = rand::thread_rng();
-    let valid_utf8: Vec<char> = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
+
+    // Primary character set (most commonly used)
+    let primary_chars: Vec<char> = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
         .chars()
         .collect();
-    let invalid_utf8 = [0xC3, 0x28]; // Invalid UTF-8 sequence
+    // Extended character set (used rarely)
+    let extended_chars: Vec<char> = (' '..=char::from_u32(0x7e).unwrap())
+        .chain((0xa0..=0x17f).filter_map(char::from_u32))
+        .filter(|c| !primary_chars.contains(c))
```

Comment:
There are no `primary_chars` in the range `0xa0` - `0x17f`
```suggestion
        .filter(|c| !primary_chars.contains(c))
        .chain((0xa0..=0x17f).filter_map(char::from_u32))
```